crates/bevy_animation/src/lib.rs

```diff
@@ -564,14 +564,14 @@ thread_local! {
 impl AnimationPlayer {
     /// Start playing an animation, restarting it if necessary.
     pub fn start(&mut self, animation: AnimationNodeIndex) -> &mut ActiveAnimation {
-        self.active_animations.entry(animation).or_default()
+        let playing_animation = self.active_animations.entry(animation).or_default();
+        playing_animation.replay();
```

Comment:
We could check if the animation matches before doing this, but I'm not actually sure if that would be faster.